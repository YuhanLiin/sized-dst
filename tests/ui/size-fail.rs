use sized_dst::*;

fn main() {
    DstA8::<dyn std::fmt::Debug, 4>::new([0u8; 17]);
}
