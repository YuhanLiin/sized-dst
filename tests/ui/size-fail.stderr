error[E0080]: evaluation of `sized_dst::assert::const_assert::<[u8; 17], sized_dst::A8, 4>::{constant#0}` failed
 --> src/assert.rs
  |
  | /         assert!(
  | |             size_of::<T>() <= N,
  | |             "Value must fit within the stack object"
  | |         );
  | |_________^ the evaluated program panicked at 'Value must fit within the stack object', $DIR/src/assert.rs:5:9
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
 --> src/assert.rs
  |
  | /     const {
  | |         assert!(
  | |             size_of::<T>() <= N,
  | |             "Value must fit within the stack object"
  | |         );
  | |     }
  | |_____^

note: the above error was encountered while instantiating `fn sized_dst::assert::const_assert::<[u8; 17], sized_dst::A8, 4>`
   --> src/lib.rs
    |
    |         assert::const_assert::<T, A, N>();
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
